src/Microsoft.Azure.SignalR.Management/README.md

```diff
@@ -0,0 +1,194 @@
+
+
+## Getting Started
+
+Azure SignalR Service Management SDK helps you to manage SignalR clients through Azure SignalR Service directly such as broadcast messages. Therefore, this SDK can be but not limited to be used in [serverless](https://azure.microsoft.com/zh-cn/solutions/serverless/) environments. You can use this SDK to manage SignalR clients connected to your Azure SignalR Service in any environment, such as in a console app, in an Azure function or in an App Server.
+
+### Features
+
+|                                     | Transient          | Persistent         |
+|-------------------------------------|--------------------|--------------------|
+| Broadcast                           | :heavy_check_mark: | :heavy_check_mark: |
+| Broadcast except some clients       | `N/A`              | :heavy_check_mark: |
+| Send to a client                    | `N/A`              | :heavy_check_mark: |
+| Send to clients                     | `N/A`              | :heavy_check_mark: |
+| Send to a user                      | :heavy_check_mark: | :heavy_check_mark: |
+| Send to users                       | `N/A`              | :heavy_check_mark: |
+| Send to a group                     | :heavy_check_mark: | :heavy_check_mark: |
+| Send to groups                      | `N/A`              | :heavy_check_mark: |
+| Send to a group except some clients | `N/A`              | :heavy_check_mark: |
+| Add a user to a group               | :heavy_check_mark: | :heavy_check_mark: |
+| Remove a user from a group          | :heavy_check_mark: | :heavy_check_mark: |
+  
+> More details about different modes can be found [here](#Transport-Type).
+
+## Quick Start
+
+### Create Service Manager
+
+Build your instance of `IServiceManager` from a `ServiceManagerBuilder`
+
+``` C#
+
+var serviceManager = new ServiceManagerBuilder()
+                    .WithOptions(option => 
+                    {
+                        option.ConnectionString = "<Your Azure SignalR Service Connection String>";
+                    })
+                    .Build();
+    
+
+```
+
+You can generate the endpoint for SignalR clients connecting to your Azure SignalR Service.
+You can also generate the access token for SignalR clients connecting to your Azure SignalR Service in serverless mode.
+
+Both of endpoint and access token are useful when you want to redirect SignalR clients to your Azure SignalR Service. The sample on how to use Management SDK to redirect SignalR clients to Azure SignalR Service can be found [here](https://github.com/aspnet/AzureSignalR-samples/tree/master/samples/Management).
```

Comment:
Explain a bit more about why you need to implement a negotiate by yourself (as there is no web server in serverless scenario).

Reply:
I think I have explained a little of it the next paragraph. The key point is the first sentence that connection string should be protected.

> Usually you don't want to expose your connection string to SignalR clients. In this situation, the SignalR clients connect to a specific endpoint which will return a special negotiation response and redirect clients to your Azure SignalR Service from the connection string. Read more details about the redirection at SignalR's [Negotiation Protocol](https://github.com/aspnet/SignalR/blob/master/specs/TransportProtocols.md#post-endpoint-basenegotiate-request).

That may not clear enough.

How about this, just move one sentence:
- - - - -
Usually you don't want to expose your connection string to SignalR clients. In this situation, the SignalR clients connect to a specific endpoint which will return a special negotiation response and redirect clients to your Azure SignalR Service from the connection string. 

Both of endpoint and access token are useful when you want to redirect SignalR clients to your Azure SignalR Service. The sample on how to use Management SDK to redirect SignalR clients to Azure SignalR Service can be found [here](https://github.com/aspnet/AzureSignalR-samples/tree/master/samples/Management).


>  Read more details about the redirection at SignalR's [Negotiation Protocol](https://github.com/aspnet/SignalR/blob/master/specs/TransportProtocols.md#post-endpoint-basenegotiate-request).
- - - - -

Reply:
I think it worth to call out in serverless scenario there is no server to do the negotiate so you need to host it by yourself. Also once you host the negotiate endpoint you can use SignalR client to connect as a normal hub.